pkg/gofr/migration/sql_test.go

```diff
@@ -4,123 +4,153 @@ import (
 	"context"
 	"database/sql"
 	"errors"
-	"reflect"
 	"testing"
 
-	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 	"go.uber.org/mock/gomock"
 
 	"gofr.dev/pkg/gofr/container"
 )
 
 func TestQuery(t *testing.T) {
 	t.Run("successful query", func(t *testing.T) {
-		ctrl := gomock.NewController(t)
-		mockDB := container.NewMockDB(ctrl)
-		expectedRows := &sql.Rows{}
+		var id int
 
-		mockDB.EXPECT().Query("SELECT * FROM users", []interface{}{}).Return(expectedRows, nil)
-		sqlMockDB := mockDB
+		var name string
 
-		rows, err := sqlMockDB.Query("SELECT * FROM users", []interface{}{})
-		if rows.Err() != nil {
-			t.Errorf("unexpected row error: %v", rows.Err())
+		expectedResult := []struct {
+			id   int
+			name string
+		}{
+			{1, "Alex"},
+			{2, "John"},
 		}
 
-		if err != nil {
-			t.Errorf("Query should return no error, got: %v", err)
-		}
+		mockContainer, mocks := container.NewMockContainer(t)
+
+		expectedRows := mocks.SQL.NewRows([]string{"id", "name"}).
+			AddRow(expectedResult[0].id, expectedResult[0].name).
+			AddRow(expectedResult[1].id, expectedResult[1].name)
+
+		mocks.SQL.ExpectQuery("SELECT * FROM users").WithoutArgs().WillReturnRows(expectedRows)
+
+		rows, err := mockContainer.SQL.Query("SELECT * FROM users")
+		require.NoError(t, err)
+
+		i := 0
 
-		if rows != expectedRows {
-			t.Errorf("Query should return the expected rows, got: %v", rows)
+		for rows.Next() {
+			err = rows.Scan(&id, &name)
+			require.NoError(t, err)
+			require.NoError(t, rows.Err())
+			assert.Equal(t, expectedResult[i].id, id)
```

Comment:
We can use replace all the `assert` with `require` so as to use only one of them for tests.